flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvFormatFactoryTest.java

```diff
@@ -227,6 +228,24 @@ public void testInvalidIgnoreParseError() {
         createTableSink(SCHEMA, options);
     }
 
+    @Test
+    public void testProjectionPushdown() throws IOException {
```

Comment:
could you add more complex test cases?

Reply:
I've added some more edge cases. Please let me know if you had something beyond that in mind - the input-output relationship seems pretty straightforward and at the moment I am lacking fantasy for more sophisticated scenarios.